--- /dev/null
+++ ../src/Models/BaseElement.php
@@ -7,6 +7,7 @@
 use DNADesign\Elemental\ORM\FieldType\DBObjectType;
 use DNADesign\Elemental\Services\ReorderElements;
 use DNADesign\Elemental\TopPage\DataExtension;
+use Psr\SimpleCache\CacheInterface;
 use Exception;
 use SilverStripe\CMS\Controllers\CMSPageEditController;
 use SilverStripe\CMS\Model\SiteTree;
@@ -607,7 +608,19 @@
         $templates = $this->getRenderTemplates();
 
         if ($templates) {
-            return $this->renderWith($templates);
+            $cache = Injector::inst()->get(CacheInterface::class . '.BaseElement');
+            $params = Controller::has_curr() ? Controller::curr()->getRequest()->getVars() : [];
+            if (isset($params['flush'])) {
+                $cache->clear();
+            }
+            unset($params['flush']);
+            $key = base64_encode($this->LastEdited . '_' . $this->ID . '_' . http_build_query($params));
+            if (!($out = $cache->get($key))) {
+                $out = $this->renderWith($templates);
+                $cache->set($key, $out);
+            }
+
+            return $out;
         }
 
         return null;
