--- /dev/null
+++ ../src/forms/HasOneAutocompleteField.php
@@ -79,7 +79,7 @@
      * @param string $sourceObject Class name of the DataObject subclass
      * @param string $labelField   The object field used for display
      */
-    public function __construct($name, $title = null, $sourceObject, $labelField = 'Title')
+    public function __construct($name, $title = null, $sourceObject = '\Page', $labelField = 'Title')
     {
         $this->sourceObject = $sourceObject;
         $this->labelField   = $labelField;
